pyanaconda/network.py

```diff
@@ -277,6 +277,78 @@ def unset(self, *args):
         ifcfglog.debug("IfcfgFile.unset %s: %s", self.filename, args)
         SimpleConfigFile.unset(self, *args)
 
+def ensure_single_initramfs_connections():
+    """Ensure device configured in initramfs has no more than one NM connection.
+
+    In case of multiple connections for device having ifcfg configuration from
+    boot options, the connection should correspond to the ifcfg file.
+    NetworkManager can be generating additional in-memory connection in case it
+    fails to match device configuration to the ifcfg (#1433891).  By
+    reactivating the device with ifcfg connection the generated in-memory
+    connection will be deleted by NM.
+
+    Don't enforce on slave devices for which having multiple connections can be
+    valid (slave connection, regular device connection).
+    """
+
+    rv = []
+
+    for dev_name in nm.nm_devices():
+        try:
+            nm.nm_device_setting_value(dev_name, "connection", "uuid")
+        except nm.SettingsNotFoundError:
+            pass
+        except nm.MultipleSettingsFoundError:
+            if nm.nm_device_is_slave(dev_name):
+                continue
+
+            ifcfg_path = find_ifcfg_file_of_device(dev_name)
+            if not ifcfg_path:
+                log.error("network: multiple settings but no ifcfg for %s", dev_name)
+                continue
+
+            # Handle only ifcfgs created from boot options in initramfs
+            # (Kickstart based ifcfgs are handled in apply_kickstart)
+            if ifcfg_is_from_kickstart(ifcfg_path):
+                continue
+
+            ensure_active_ifcfg_connection_for_device(ifcfg_path, dev_name, only_replace=True)
+            rv.append(dev_name)
```

Comment:
I've already split out ensure_active_ifcfg_connection_for_device from the function during rhel7-branch patch review. I can't see a reason for going further and having something like _handle_multiple_setting_from_initramfs().